### ISSUE TITLE
feat(MLOps): Implement VideoMAE Backbone & Binary Classification Head

### ISSUE THREAD

**Issue by mauriziokeningen** [mauriziokeningen/Computer-Vision-Residential-Security-System#56]
# feat(MLOps): Implement VideoMAE Backbone & Binary Classification Head

**Epic:** Epic 3 (Model Architecture)
**Assignee:** @mauriziokeningen 
**Priority:** High 
**Status:** To Do
**Labels:** `machine-learning`, `model-arch`, `pytorch`

### Background & Objective
We are transitioning our perception pipeline from general-purpose action recognition to domain-specific anomaly detection. The objective of this task is to ingest the `MCG-NJU/videomae-base` pre-trained weights and surgically replace the 400-class Kinetics head with a 2-class binary head (`Threat` vs. `Decoy`). 

This component must be production-ready: statically typed, logged, and decoupled from execution scripts so it can be easily imported into the upcoming PyTorch Lightning training loop.

### Acceptance Criteria
- [ ] **Dependency Management:** Hugging Face `transformers` and `torch` are integrated into the `pyproject.toml` / Poetry environment.
- [ ] **Safe Weight Acquisition:** Model downloads and caches weights strictly to the 1.6TB partition (`$HF_HOME` is respected; absolutely no weights land on the root partition).
- [ ] **Architecture Modification:** The base VideoMAE graph is preserved, but the `classifier` layer is replaced with a `Linear(in_features, 2)` layer.
- [ ] **Production Standards:** Code implements standard Python `logging`, explicit type hints (`typing`), and avoids magic numbers.
- [ ] **Automated Smoke Test:** A dedicated unit test (`tests/test_model.py`) executes a forward pass with a dummy tensor `[4, 3, 16, 224, 224]` and asserts the output shape is `(4, 2)`.
- [ ] **Hardware Audit:** The forward pass successfully executes on the RTX 4090 without triggering a CUDA Out-of-Memory (OOM) exception.

---

### Architectural Guidelines (Tech Lead Notes)

Do not script this; architect it. We need a clean, importable function and a separate testing block. Follow this architectural blueprint:

**1. Core Implementation (`research/src/model.py`):**
```python
import logging
import torch
from typing import Tuple
from transformers import VideoMAEForVideoClassification, PreTrainedModel

logger = logging.getLogger(__name__)

def get_residential_videomae(
    model_name: str = "MCG-NJU/videomae-base", 
    num_classes: int = 2
) -> Tuple[PreTrainedModel, torch.device]:
    """
    Initializes the VideoMAE backbone and swaps the classification head for binary threat detection.
    """
    logger.info(f"Mounting {model_name} with {num_classes} output classes...")
    
    # ignore_mismatched_sizes=True safely discards the old 400-class head
    model = VideoMAEForVideoClassification.from_pretrained(
        model_name,
        num_labels=num_classes,
        ignore_mismatched_sizes=True
    )
    
    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
    model.to(device)
    
    logger.info(f"Architecture successfully mounted to {device}.")
    return model, device
```

**2. The Tracer Bullet (`research/tests/test_model.py`):**
```python
import torch
import logging
from src.model import get_residential_videomae

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

def test_videomae_forward_pass():
    """Smoke test to ensure the head was replaced and tensors flow correctly."""
    model, device = get_residential_videomae()
    
    # Simulated input: Batch=4, Channels=3, Frames=16, H=224, W=224
    dummy_input = torch.randn(4, 3, 16, 224, 224).to(device)
    
    logger.info("Executing tracer bullet forward pass...")
    with torch.no_grad():
        outputs = model(dummy_input)
        logits = outputs.logits
        
    assert logits.shape == (4, 2), f"FATAL: Expected shape (4, 2), got {logits.shape}"
    logger.info(f"Tracer Bullet Success. Logit Shape: {logits.shape}")

if __name__ == "__main__":
    test_videomae_forward_pass()
```
